src/shared/specifier.c

```diff
@@ -67,6 +67,12 @@ int specifier_printf(const char *text, size_t max_length, const Specifier table[
                                         size_t k, j;
 
                                         r = i->lookup(i->specifier, i->data, root, userdata, &w);
+                                        if (r == -ENOENT)
+                                                /* Many specifiers refer to machine-id or os-release fields,
+                                                 * which will return -ENOENT if the file is not present.
+                                                 * Let's translate this to -ENODATA to make the error more
+                                                 * understandable in the lower layers of the call stack. */
+                                                return -ENODATA;
```

Comment:
hmm, shouldn't the individual specifier callbcks already convert that? Sounds weird to convert the errors so late... And shouldn't this be a bit more exotic errno? i.e. EUNATCH maybe?